pkg/util/validation/limits.go

```diff
@@ -228,6 +250,20 @@ func (l *Limits) Validate(shardByAllLabels bool) error {
 		return errMaxGlobalSeriesPerUserValidation
 	}
 
+	// If query priority is enabled, do not allow duplicate priority values
+	if l.QueryPriority.Enabled {
```

Comment:
Just FYI, seems this `Lists.Validate` method is only called when starting Cortex.
At runtime, when hot reloading configs, this method is not called. Maybe we should add that.

Reply:
> But this function is called every time user makes a query call, are we ok with validating all the time? Or is there a method that gets called only when there is a change in the config (yaml)?

I guess we can still do it the same place when we unmarshal the yaml/json. But the downside is that it is hard to track whether config changes or not.

The default runtime reload interval is 10s. I think it is fine to check it at this interval. WDYT? We can combine with LRU cache also.

Reply:
Addressed in https://github.com/cortexproject/cortex/pull/5605/commits/22be0b8a827db870eff985ba56c1bd2807b3107a